src/codemagic/tools/app_store_connect.py

```diff
@@ -492,25 +492,53 @@ def list_devices(
         "register-device",
         BundleIdArgument.PLATFORM,
         DeviceArgument.DEVICE_NAME,
-        DeviceArgument.DEVICE_UDID,
+        DeviceArgument.DEVICE_UDIDS,
+        DeviceArgument.IGNORE_REGISTRATION_ERRORS,
     )
     def register_device(
         self,
         platform: BundleIdPlatform,
         device_name: str,
-        device_udid: str,
+        device_udids_argument: Union[str, Types.DeviceUdidsArgument, Sequence[Union[str, Types.DeviceUdidsArgument]]],
+        ignore_registration_errors: bool = False,
         should_print: bool = True,
-    ) -> Device:
+    ) -> List[Device]:
         """
-        Register a new device for app development
+        Register new Devices for app development
         """
-        return self._create_resource(
-            self.api_client.devices,
-            should_print,
-            udid=device_udid,
-            name=device_name,
-            platform=platform,
-        )
+        if isinstance(device_udids_argument, str) or isinstance(device_udids_argument, Types.DeviceUdidsArgument):
```

Comment:
It should be checked that `device_udids_argument` is not `None`, which it can be as `DeviceArgument.DEVICE_UDIDS` is optional.